src/main/java/picard/analysis/TheoreticalSensitivity.java

```diff
@@ -204,4 +208,187 @@ public int draw() {
+     * @param alleleFraction the allele fraction to evaluate sensitivity at
+     * @return Theoretical sensitivity for the given arguments over a particular depth distribution.
+     */
+    public static double theoreticalSensitivity(final Histogram<Integer> depthHistogram, final Histogram<Integer> qualityHistogram,
+                                                final int sampleSize, final double logOddsThreshold, final double alleleFraction) {
+        if (alleleFraction > 1.0 || alleleFraction < 0.0) {
+            throw new IllegalArgumentException("Allele fractions must be between 0 and 1.");
+        }
+
+        final double[] depthDistribution = normalizeHistogram(depthHistogram);
+
+        // Integrate sensitivity over depth distribution
+        double sensitivity = 0.0;
+        int currentDepth = 0;
+        double right = 0;
+        while (currentDepth < depthDistribution.length) {
+            double deltaDepthProbability = 0.0;
+            // Accumulate a portion of the depth distribution to compute theoretical sensitivity over.
+            // This helps prevent us from spending lots of compute over coverages
+            // that occur with low probability and don't contribute much to sensitivity anyway, but
+            // it complicates things a bit by having a variable deltaDepthProbability which
+            // amount of the depth distribution to use with the trapezoid rule integration step.
+            while (deltaDepthProbability == 0 && currentDepth < depthDistribution.length ||
+                    deltaDepthProbability < DEPTH_BIN_WIDTH && currentDepth < depthDistribution.length &&
+                    depthDistribution[currentDepth] < DEPTH_BIN_WIDTH / 2.0) {
+                deltaDepthProbability += depthDistribution[currentDepth];
+                currentDepth++;
+            }
+            // Calculate sensitivity for a particular depth, and use trapezoid rule to integrate sensitivity.
+            final double left = right;
+            right = sensitivityAtConstantDepth(currentDepth, qualityHistogram, logOddsThreshold, sampleSize, alleleFraction);
+            sensitivity += deltaDepthProbability * (left + right) / 2.0;
+        }
+        return sensitivity;
+    }
+
+    /**
+     * Removes trailing zeros in a distribution.  The purpose of this function is to prevent other
+     * functions from evaluating in regions where the distribution has zero probability.
+     * @param distribution Distribution of base qualities
+     * @return Distribution of base qualities removing any trailing zeros
+     */
+     static double[] trimDistribution(final double[] distribution) {
+        int endOfDistribution = 0;
+
+        // Locate the index of the distribution where all the values remaining at
+        // larger indices are zero.
+        for (endOfDistribution = distribution.length - 1; endOfDistribution >= 0; endOfDistribution--) {
```

Comment:
What about
```
int end = distribution.length;
while(distribution[end] != 0) {
    end--;
}
```
Or even shorter
```
int end = distribution.length;
while( distribution[--end] != 0) { }
```
or
```
final int numTrailingZeroes = IntStream.range(1, distribution.length).findFirst(n -> distribution[distribution.length - n] != 0);
return Arrays.copyOfRange(distribution, 0, distribution.length = numTrailingZeroes);
```

Reply:
Not sure how important this is, but these approaches result in copying the entire array if all elements are zero.

Reply:
fixed